internal/core/workers/operation_execution_worker/operation_execution_worker.go

```diff
@@ -76,125 +79,158 @@ func (w *OperationExecutionWorker) Start(ctx context.Context) error {
 			if errors.Is(err, context.Canceled) {
 				return nil
 			}
-
-			w.Stop(ctx)
 			reportOperationExecutionWorkerFailed(w.scheduler.Game, w.scheduler.Name, LabelNextOperationFailed)
 			return fmt.Errorf("failed to get next operation: %w", err)
 		}
 
-		loopLogger := w.logger.With(
-			zap.String(logs.LogFieldOperationID, op.ID),
-			zap.String(logs.LogFieldOperationDefinition, def.Name()),
-		)
-
-		if op.Status != operation.StatusPending {
-			loopLogger.Warn("operation is at an invalid status to proceed")
-
-			continue
-		}
-
-		executor, hasExecutor := w.executorsByName[def.Name()]
-		if !hasExecutor {
-			loopLogger.Warn("operation definition has no executor")
-
-			w.evictOperation(ctx, loopLogger, op)
-			reportOperationEvicted(w.scheduler.Game, w.scheduler.Name, op.DefinitionName, LabelNoOperationExecutorFound)
-
-			continue
+		err = w.executeOperation(op, def)
+		if err != nil {
+			switch {
+			case errors.Is(err, workererrors.ErrStartOperationFailed) ||
```

Comment:
Now it is easier to decide if we should stop the worker in those scenarios. Initially, I made just a refactor, didn't change the logic at all... Now it is a chance to change this behavior, @arthur29 @caiordjesus what do you guys think? should we continue to stop the worker and let the worker manage to recreate it in those scenarios? Do you guys think it is ok to change this behavior is this PR?